presto-main/src/main/java/io/prestosql/sql/planner/plan/ApplyNode.java

```diff
@@ -95,6 +96,14 @@ public ApplyNode(
         this.subqueryAssignments = subqueryAssignments;
         this.correlation = ImmutableList.copyOf(correlation);
         this.originSubquery = originSubquery;
+
+        ImmutableList.Builder<Symbol> outputSymbols = ImmutableList.builder();
+        outputSymbols.addAll(input.getOutputSymbols());
+        subqueryAssignments.getOutputs()
+                .stream()
+                .filter(symbol -> !input.getOutputSymbols().contains(symbol))
```

Comment:
`input.getOutputSymbols()` returns a list, so it is quadratic

Reply:
I've opened separate PR https://github.com/prestosql/presto/pull/462 for the first two commits and addressed this comment there